### PR TITLE
Add possibility to override project config #1891

### PR DESCRIPTION
**Pull request by staifa**
Please answer the following questions and leave the below in as part of your PR.

- [X] I have read the [developer documentation](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md).

- [X] This PR corresponds to an [issue with a clear problem statement](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md#start-with-an-issue-before-writing-code).

- [X] This PR contains a [test](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md#tests) to prevent against future regressions

- [X] I have updated the [CHANGELOG.md](https://github.com/clj-kondo/clj-kondo/blob/master/CHANGELOG.md) file with a description of the addressed issue.

Closes https://github.com/clj-kondo/clj-kondo/issues/1891

### CODE REVIEW

**Comment by staifa** [comment]
@borkdude set to draft, I realized that `:extra-config-path` overrides command line config and namespace local config too. I'll fix it. How do you feel about overriding the command line config? It may be a good thing in my opinion - you don't have control over it when using `clojure-lsp` for example. The user should be able to completely configure his dev env...

**Comment by borkdude** [comment]
> you don't have control over it when using clojure-lsp for example. The user should be able to completely configure his dev env

yes, but then again, the developer is going to provide extra config via the command line as well, which then should override the extra-configs ;)

I don't think it will often be necessary to override the command line config of clojure-lsp, right? Do you have a good example of that?

**Comment by staifa** [comment]
I don't :) No overriding command line config then!

**Comment by staifa** [comment]
Ready for review @borkdude

**Comment by borkdude** [comment]
Can you make another dummy commit? I'm not able to trigger CI via the admin interface.

File: src/clj_kondo/impl/core.clj [hunk 1]
```diff
@@ -61,10 +61,9 @@
 (defn opts [^java.io.File cfg-file]
   (let [include #(read-fn cfg-file read-edn-file %)]
     {:readers
-     {'include include
+     {'include include}}))
```
Comment (borkdude) [inline src/clj_kondo/impl/core.clj:64]: Can you revert this? I'd like to keep those comments within the map 🤣

Comment (staifa) [reply src/clj_kondo/impl/core.clj]: Sorry, parinfer :)

**Comment by staifa** [comment]
Done.

**Comment by borkdude** [comment]
I'm not sure what's up with CircleCI here, but I'll get back to this PR in the next few days.

One thought I had: maybe it would be nice to be able to provide a config override that doesn't have to be visible in the checked-in code? E.g. via en environment variable or so?

**Comment by staifa** [comment]
> I'm not sure what's up with CircleCI here, but I'll get back to this PR in the next few days.
> 
> One thought I had: maybe it would be nice to be able to provide a config override that doesn't have to be visible in the checked-in code? E.g. via en environment variable or so?

Neat idea, I pushed a commit implementing env var config paths.

File: doc/config.md [hunk 2]
```diff
@@ -51,6 +51,11 @@ Clj-kondo can be configured in several ways:
 - project config: a `config.edn` file in the `.clj-kondo` directory (see
   [project setup](../README.md#project-setup)).
 - `:config-paths` in project `config.edn`: a list of directories that provide additional config.
+- `:extra-config-paths` in project `config.edn`: a list of directories that provide additional config
```
Comment (borkdude) [inline doc/config.md:54]: Should we still have the `:extra-config-paths` if we have the environment variable?

Is it easy to set the environment variable for you editor on a project by project basis, have you tested this?

**Comment by staifa** [comment]
Ok, only env var handling is implemented now. I hand tested ant everything works as expected. I removed tests for now - I noticed you don't use `with-redefs`, how would you test setting and unsetting the env variable?

**Comment by borkdude** [comment]
I feel like we need to think a little more about this one in the coming weeks, so I haven't included in the current release yet, but I will come back to this.

**Pull request closed without merging**